# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/04 21:28:41-08:00 akpm@osdl.org 
#   [PATCH] suppress might_sleep() if oopsing
#   
#   We can call might_sleep() functions on the oops handling path (under do_exit).
#   
#   There seem little point in emitting spurious might_sleep() warnings into the
#   logs after the kernel has oopsed.
#   
#   Signed-off-by: Andrew Morton <akpm@osdl.org>
#   Signed-off-by: Linus Torvalds <torvalds@osdl.org>
# 
# kernel/sched.c
#   2005/01/04 20:24:26-08:00 akpm@osdl.org +1 -1
#   suppress might_sleep() if oopsing
# 
diff -Nru a/kernel/sched.c b/kernel/sched.c
--- a/kernel/sched.c	2005-01-22 22:57:56 -08:00
+++ b/kernel/sched.c	2005-01-22 22:57:56 -08:00
@@ -4631,7 +4631,7 @@
 	static unsigned long prev_jiffy;	/* ratelimiting */
 
 	if ((in_atomic() || irqs_disabled()) &&
-	    system_state == SYSTEM_RUNNING) {
+	    system_state == SYSTEM_RUNNING && !oops_in_progress) {
 		if (time_before(jiffies, prev_jiffy + HZ) && prev_jiffy)
 			return;
 		prev_jiffy = jiffies;
